internal/proto/readers/v1/readers.proto

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package readers.v1;
+
+option go_package = "github.com/absmach/magistrala/internal/grpc/readers/v1";
+
+// ReadersService is a service that provides access to
+// readers functionalities for Magistrala services.
+service ReadersService {
+  rpc ReadMessages(ReadMessagesReq)
+    returns (ReadMessagesRes) {}
+}
+
+message PageMetadata {
+  uint64 limit               = 1;
+  uint64 offset              = 2;
+  string protocol            = 3;
+  string name                = 4;
+  double value               = 5;
+  string publisher           = 6;
+  bool bool_value            = 7;
+  string string_value        = 8;
+  string data_value          = 9;
+  double from                = 10;
+  double to                  = 11;
+  string subtopic            = 12;
+  string interval            = 13;
+  bool read                  = 14;
+  Aggregation aggregation    = 15;
+  string comparator          = 16;
+  string format              = 17;
+}
+
+message ReadMessagesRes {
+  uint64 total                        = 1;
+  PageMetadata page_metadata          = 2;
+  repeated Message messages           = 3;
+}
+
+message Message {
+  oneof payload {
+    SenMLMessage senml = 1;
+    JsonStructuredMessage json = 2;
+  }
+}
+
+message SenMLMessage {
+  string channel = 1;
+  string subtopic = 2;
+  string publisher = 3;
+  string protocol = 4;
+  string name = 5;
+  string unit = 6;
+  double time = 7;
+  double update_time = 8;
+  double value = 9;
+  string string_value = 10;
+  string data_value = 11;
+  bool bool_value = 12;
+  double sum = 13;
+}
+
+message JsonStructuredMessage {
+  string channel = 1;
```

Comment:
we can have abstract of following , because they will be same for both senML and json messages 
```proto
  string channel = 1;
  string subtopic = 2;
  string publisher = 3;
  string protocol = 4;
```